.github/workflows/FormatCheck.yml

```diff
@@ -58,5 +59,9 @@ jobs:
           end
           if !formatted
               @error "Found some unformatted files:\n" * read(`git diff --name-only`, String)
-              exit(1)
           end
+      - uses: reviewdog/action-suggester@v1
+        if: github.event_name == 'pull_request'
+        with:
+          tool_name: JuliaFormatter
+          fail_on_error: true
```

Comment:
It would be nice if the format workflow would run all the time but just avoid doing suggestions until out of draft mode. This could look something like this:
```yaml
jobs:
  format-check:
    env:
      add_suggestions: ${{ github.event_name == 'pull_request' && github.event.pull_request.draft == false }}
    steps:
      - name: Check formatting
        run: |
          ...
          if !formatted
              @error "Found some unformatted files:\n" * read(`git diff --name-only`, String)
              get(ENV, "add_suggestions", "false") == "false" && exit(1)
          end
      - uses: reviewdog/action-suggester@v1
        if: ${{ env.add_suggestions }}
        with:
          tool_name: JuliaFormatter
          fail_on_error: true
```

Reply:
The problem with the current setup is that a formatting error to `main` won't cause a workflow failure

Reply:
Another option I like better:
```yaml
jobs:
  format-check:
    steps:
      - name: Check formatting
        id: check
        run: |
          ...
          if !formatted
              @error "Found some unformatted files:\n" * read(`git diff --name-only`, String)
          end
          open(ENV["GITHUB_OUTPUT"], "a") do io
              println(io, "formatted=$formatted")
          end
      - uses: reviewdog/action-suggester@v1
        if: ${{ github.event_name == 'pull_request' && github.event.pull_request.draft == false }}
        with:
          tool_name: JuliaFormatter
          fail_on_error: true
      - name: Fail if not formatted
        if: ${{ steps.check.outputs.formatted != 'true' }}
        run: exit 1
```